src-ext/TypeScript/ggvega/src/itmLayerArrayByScales.ts

```diff
@@ -21,6 +23,24 @@ import {ItmLayer} from './itmLayer';
  * @returns `ItmLayer[]`
  */
 export function itmLayerArrayByScalesArray(itmLayerArray: ItmLayer[], ggScaleArray: GG.Scale[]): ItmLayer[] {
+  itmLayerArray.map((itmLayer: ItmLayer) => {
+    for (const encodingKey in itmLayer.encoding) {
+      ggScaleArray.map((ggScale: GG.Scale) => {
+        //NOTE @wenyu:https://love2dev.com/blog/javascript-remove-from-array/
+
+        for (let i = 0; i < ggScale.aesthetics.length; i++) {
+          if (ggScale.aesthetics[i] === encodingKey) {
+            itmLayer.encoding[encodingKey].title = ggScale.name;
+            if (ggScale.class === 'ScaleContinuousPosition') {
```

Comment:
we should use the class dispatch method here, like outlined in Facet or Stat - this will require you to update `Scale` in ggschema to support classes like `ScaleContinuousPosition`.

Reply:
done